client/my-sites/site-settings/site-icon-setting/site-icon-media-modal/index.jsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import AsyncLoad from 'components/async-load';
+import Dialog from 'components/dialog';
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+class SiteIconMediaModal extends Component {
```

Comment:
I think I'd anticipated there being quite a bit of logic here, though it doesn't seem separate in purpose than `<SiteIconSetting />`. I'll see if I can consolidate.